src/main/java/com/faforever/api/mautic/MauticService.java

```diff
@@ -56,22 +62,27 @@ public MauticService(MappingJackson2HttpMessageConverter mauticApiMessageConvert
     // TODO for now, client ID needs to be a username and client secret the user's password.
     RestTemplateBuilder builder = restTemplateBuilder.basicAuthentication(mauticProperties.getClientId(), mauticProperties.getClientSecret());
 
-    restOperations = builder.build();
+    restTemplate = builder.build();
   }
 
-  @Async
-  public CompletableFuture<Object> createOrUpdateContact(String email, String fafUserId, String fafUserName, String ipAddress, OffsetDateTime lastActive) {
-    Map<String, Object> body = new HashMap<>();
-
-    // These are Mautic default fields. For some reason, these are camel case.
-    Optional.ofNullable(email).ifPresent(s -> body.put("email", email));
-    Optional.ofNullable(ipAddress).ifPresent(s -> body.put("ipAddress", ipAddress));
-    Optional.ofNullable(lastActive).ifPresent(s -> body.put("lastActive", lastActive));
+  @Override
+  public void userDataChanged(UserUpdatedEvent event) {
+    Map<String, Object> body = Map.of(
+      // These are Mautic default fields. For some reason, these are camel case.
+      "email", event.getEmail(),
+      "ipAddress", event.getIpAddress(),
+      "lastActive", OffsetDateTime.now(),
 
-    // These are Mautic "custom fields" that need to be created  explicitly. For some reason, these are underscore case by default.
-    Optional.ofNullable(fafUserId).ifPresent(s -> body.put("faf_user_id", fafUserId));
-    Optional.ofNullable(fafUserName).ifPresent(s -> body.put("faf_username", fafUserName));
+      // These are Mautic "custom fields" that need to be created  explicitly. For some reason, these are underscore case by default.
+      "faf_user_id", event.getId(),
+      "faf_username", event.getUsername()
+    );
 
-    return CompletableFuture.completedFuture(restOperations.postForObject("/contacts/new", body, Object.class));
+    try {
+      restTemplate.postForObject("/contacts/new", body, Object.class);
+      log.debug("Updated contact in Mautic from event: {}", event);
+    } catch (Exception e) {
+      log.error("Could not update contact in Mautic for user id: {}", event.getId(), e);
```

Comment:
error is supposed to be logged if application gets unfunctional by error, so maybe this would might rather be warn(maybe mautic is down)...

Reply:
This is exactly the point. If Mautic is down, faf and mautic go out of sync , which is an error and should be fixed (there is no way to do it right now).